File: Practica_4/civitasJava/src/civitas/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */ 
package civitas;

import java.util.ArrayList;
 
/**
 *
 * @author mariolopezgonzalez
 */
public class CasillaJuez extends Casilla{
     
    private static int carcel = 1;

    /**
     * Constructor de casillas de tipo JUEZ
     */
    CasillaJuez(int numCasillaCarcel, String nombre) {
        super(nombre);
        carcel = numCasillaCarcel;
    }

    
    /**
     * Metodo recibeJugador_juez
     *  Parámetros:
     *    iactual = indice del jugador actual
     *    todos = Lista con todos los jugadores
     *  Efecto:
     *    La casilla de tipo juez recibe al jugador actual al que se le aplica el efecto de ésta 
     *    y se informa al diario.
     */
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(iactual, todos)){
            informe(iactual, todos);
            todos.get(iactual).encarcelar(carcel);
        }
    }

    /**
     * Sobrecarga del metodo toString() de java
     */
    @Override
    public String toString() {
        return "Casilla:  nombre = " + super.getNombre() + " numero de la Casilla Carcel = " + carcel;
    }
}

File: Practica_4/civitasRuby/lib/casilla_juez.rb
#encoding:utf-8

module Civitas
  class Casilla_juez < Casilla
    @@carcel = 1

    def initialize (num_casilla_carcel, nombre)
       super(nombre)
       @@carcel = num_casilla_carcel
    end

    #  Metodo recibe_jugador
    #   Parámetros:
    #     iactual = indice del jugador actual
    #     todos = Lista con todos los jugadores
    #   Efecto: 
    #     La casilla de tipo juez recibe al jugador actual al que se le aplica el efecto de ésta 
    #     y se informa al diario.
    def recibe_jugador(iactual, todos)
      if jugador_correcto(iactual, todos)
        informe(iactual, todos)
        todos[iactual].encarcelar(@@carcel)
      end
    end

    # Sobrecarga del metodo to_s de Ruby
    def to_s
      return "Casilla Juez -> Nombre: #{nombre}, Tipo: #{@tipo}, Número Cárcel: #{@@carcel}"
    end
  end
end
